packages/std/src/math.rs

```diff
@@ -180,6 +180,100 @@ impl Uint128 {
     pub fn is_zero(&self) -> bool {
         self.0 == 0
     }
+
+    pub fn checked_add(self, other: Self) -> StdResult<Self> {
+        self.0
+            .checked_add(other.0)
+            .map(Self)
+            .ok_or_else(|| StdError::overflow(self, other))
```

Comment:
Can you create a specific

```rust
struct Uint128Overflow {
   // all the error information we need
}
```

and implement `From<Uint128Overflow> for StdError`?

This would make clear there is only one error type that can happen (as expressed by the Option API from Rust) but at the same time make it easy for contract developers to just use `?`.

Reply:
done ;D